### ISSUE TITLE
Override isValidMove(const Move& move) for Pawn class

### ISSUE THREAD

**Issue by finicu212** [finicu212/openChess#13]
Do not treat:

- en passant

(make separate PR later)

- [x] Don't move if will be in check EDIT: to be handled in a separate PR

- [x] Capturing logic

- [x] Moving logic

- [x] pawn promotion, automatically make queen for now

To be done only after #24 has been merged
